.github/workflows/tests.yml

```diff
@@ -0,0 +1,30 @@
+name: Unit Tests
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  test:
+    name: Run Unit Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up shell environment
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y shellcheck
+        shell: bash
+
+      - name: Run shellcheck
+        run:  shellcheck --severity warning scripts/upgrade.sh scripts/tests.sh
+
+      - name: Run unit tests
+        run: |
+          cd scripts
+          chmod +x upgrade.sh tests.sh
+          ./tests.sh
```

Comment:
You could just make these files executable in the repo, instead of chmod'ing them here.

Also, this test execution should run in a busybox image, to ensure that we're using the same interpreter as the shipping docker image.